cli/internal/run/run.go

```diff
@@ -663,17 +662,18 @@ func (r *run) logWarning(prefix string, err error) {
 }
 
 func (r *run) executeTasks(ctx gocontext.Context, g *completeGraph, rs *runSpec, engine *core.Scheduler, packageManager *packagemanager.PackageManager, hashes *taskhash.Tracker, startAt time.Time) error {
+	apiClient := r.config.NewClient()
 	var analyticsSink analytics.Sink
 	if r.config.IsLoggedIn() {
-		analyticsSink = r.config.ApiClient
+		analyticsSink = apiClient
```

Comment:
This is starting to make me consider threading a context through the client.

Reply:
I think that would probably be a good thing to do. We don't fully take advantage of `Context` now, but http requests are a natural place to do so.

It might be somewhat tricky with the analytics client, since it does batching and flushing in a goroutine, so you can't directly use a `Context` from anyone who wants to record an event, but there's probably a way to make it work out.